src/debug/debug.cc

```diff
@@ -352,10 +361,55 @@ DebugStream() {
 static void
 ResetSections(const int level)
 {
+    DidResetSections = true;
     for (auto &sectionLevel: Debug::Levels)
         sectionLevel = level;
 }
 
+/// optimization: formats ProcessLabel once for frequent debugs() reuse
+static void
+LabelThisProcess(const char * const name, const Optional<int> id = Optional<int>())
+{
+    assert(name);
+    assert(strlen(name));
+    std::stringstream os;
+    os << ' ' << name;
+    if (id.has_value()) {
+        assert(id.value() >= 0);
+        os << id.value();
+    }
+    ProcessLabel = os.str();
+}
+
+void
+Debug::NameThisHelper(const char * const name)
+{
+    LabelThisProcess(name);
+
+    if (const auto parentProcessDebugOptions = getenv("SQUID_DEBUG")) {
+        assert(!debugOptions);
+        debugOptions = xstrdup(parentProcessDebugOptions);
+    }
+
+    // do not restrict helper (i.e. stderr) logging beyond debug_options
+    EnsureDefaultStderrLevel(DBG_DATA);
+
+    BanCacheLogUse();
+    SettleStderr();
+    SettleSyslog();
```

Comment:
FYI; The helpers we bundle should not be using syslog. They should only be using stderr currently.

Please ensure that non-use for consistency until the other helpers are updated to this new debugs API. Then we can do the switch to enable it for all of them at some easily documented vNN release.

Reply:
> Please ensure that non-use for consistency until the other helpers are updated to this new debugs API.

I believe the proposed code does not enable syslog use by helpers: SettleSyslog() does not enable syslog use by default, and existing helpers do not call Debug::ConfigureSyslog() to overwrite that default. I see no pinger messages in syslog when testing this PR.